src/main/java/org/opensearch/neuralsearch/ml/MLCommonsClientAccessor.java

```diff
@@ -262,4 +314,52 @@ private MLInput createMLMultimodalInput(final List<String> targetResponseFilters
         final MLInputDataset inputDataset = new TextDocsInputDataSet(inputText, modelResultFilter);
         return new MLInput(FunctionName.TEXT_EMBEDDING, null, inputDataset);
     }
+
+    /**
+     * Retryable method for sentence highlighting inference.
+     * This method handles retries and calls the ML client with the appropriate input.
+     *
+     * @param inferenceRequest The sentence highlighting inference request
+     * @param retryTime Current retry count
+     * @param listener ActionListener to handle the response or failure
+     */
+    private void retryableInferenceSentenceHighlighting(
+        final SentenceHighlightingRequest inferenceRequest,
+        final int retryTime,
+        final ActionListener<List<Map<String, Object>>> listener
+    ) {
+        try {
+            MLInputDataset inputDataset = new QuestionAnsweringInputDataSet(inferenceRequest.getQuestion(), inferenceRequest.getContext());
+            MLInput mlInput = new MLInput(FunctionName.QUESTION_ANSWERING, null, inputDataset);
+
+            mlClient.predict(inferenceRequest.getModelId(), mlInput, ActionListener.wrap(mlOutput -> {
+                List<Map<String, Object>> result = processHighlightingOutput(mlOutput);
```

Comment:
You need to catch exception.
try {
  List<Map<String, Object>> result = processHighlightingOutput(mlOutput);
  listener.onRespone(result);
} catch (Exception e) {
  listener.onFailure(e);

}

Reply:
Ack and updated.